### PR TITLE
 refactor(artifacts): Add customKind flag to artifact config

### PR DESCRIPTION
**Pull request by ezimanyi**

### CODE REVIEW

File: app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts [hunk 1]
```diff
@@ -35,33 +35,34 @@ describe('ExpectedArtifactService', () => {
         isDefault: true,
       },
     ].map(k => ({ ...baseKindConfig, ...k }));
-    const defaultKindConfig = {
+    const customKindConfig = {
       ...baseKindConfig,
       key: 'custom',
+      isCustom: true,
```
Comment (sbwsg) [inline app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts]: Is there a mismatch here or is this property name intentional? `isCustom` here vs `customKind` elsewhere.

File: app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts [hunk 2]
```diff
@@ -87,15 +88,15 @@ describe('ExpectedArtifactService', () => {
...
     it('returns the default kind if neither kind nor type are stored on artifact when isDefault is true', () => {
       const artifact: IArtifact = {
         id: 'artifact-id',
       };
       const kindConfig = ExpectedArtifactService.getKindConfig(artifact, true);
-      expect(kindConfig).toEqual(defaultKindConfig);
+      expect(kindConfig).toEqual(customKindConfig);
     });
   });
 });
```
Comment (sbwsg) [inline app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts]: Would it be worth adding a test ensuring that an artifact with no `kind` field but a `customKind` flag w/ a known type correctly results in `getKindConfig` returning the `customKindConfig`? E.g. something like:

```js
it('returns the default kind if an artifact has a customKind flag, regardless of its type', () => {
  const artifact: IArtifact = {
    id: 'artifact-id',
    customKind: true,
    type: 'foo-type'
  };
  const kindConfig = ExpectedArtifactService.getKindConfig(artifact, false);
  expect(kindConfig).toEqual(customKindConfig);
});
```

File: app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts (hunk 1 above)
Comment (ezimanyi) [reply app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts]: Good catch, I renamed it and missed this test

File: app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts (hunk 2 above)
Comment (ezimanyi) [reply app/scripts/modules/core/src/artifact/expectedArtifact.service.spec.ts]: Yes, good point; I'll add such a test

**Pull request merged**